python/cudf/cudf/core/column/string.py

```diff
@@ -493,7 +493,10 @@ def _memory_usage(self, deep=False):
             return self.str().size() * self.dtype.itemsize
 
     def __len__(self):
-        return self.nvstrings.size()
+        if self.size:
+            return len(self.children[0]) - 1
+        else:
+            return 0
```

Comment:
We can just return `self.size` here in general instead of reaching into the children unnecessarily. Size is an attribute of the column.

Reply:
Oh sorry I misunderstood. So `self.size` is the number of strings?

Reply:
A column's size is the number of elements. In the case of a strings column, the elements are strings.

Reply:
Yeah sorry I think I just confused myself. This isn't a part of the code I'm particularly familiar with.

Should be fixed in the latest commit 🙂